contrib/geoJSON_export.lua

```diff
@@ -0,0 +1,320 @@
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+]]
+--[[
+darktable geoJSON export script
+
+ADDITIONAL SOFTWARE NEEDED FOR THIS SCRIPT
+* mkdir
+* convert (ImageMagick)
+* xdg-open
+* xdg-user-dir
+
+WARNING
+This script is only tested with Linux
+
+USAGE
+* require this script from your main Lua file
+
+]]
+   
+local dt = require "darktable"
+local gettext = dt.gettext
+dt.configuration.check_version(...,{3,0,0})
+	
+-- Tell gettext where to find the .mo file translating messages for a particular domain
+gettext.bindtextdomain("geoJSON_export",dt.configuration.config_dir.."/lua/")
+
+local function _(msgid)
+    return gettext.dgettext("geoJSON_export", msgid)
+end
+
+-- Sort a table
+local function spairs(_table, order) -- Code copied from http://stackoverflow.com/questions/15706270/sort-a-table-in-lua
+    -- collect the keys
+    local keys = {}
+    for _key in pairs(_table) do keys[#keys + 1] = _key end
+
+    -- if order function given, sort by it by passing the table and keys a, b,
+    -- otherwise just sort the keys 
+    if order then
+        table.sort(keys, function(a,b) return order(_table, a, b) end)
+    else
+        table.sort(keys)
+    end
+
+    -- return the iterator function
+    local i = 0
+    return function()
+        i = i + 1
+        if keys[i] then
+            return keys[i], _table[keys[i]]
+        end
+    end
+end
+
+local function show_status(storage, image, format, filename, number, total, high_quality, extra_data)
+    dt.print(string.format(_("Export Image %i/%i"), number, total))
+end
+
+local function checkIfBinExists(bin)
+    local handle = io.popen("which "..bin)
+    local result = handle:read()
+    local ret
+    handle:close()
+    if (not result) then
+        dt.print_error(bin.." not found")
+        ret = false
+    end
+    ret = true
+    return ret
+end
+
+local function create_geoJSON_file(storage, image_table, extra_data)
+
+    if not checkIfBinExists("mkdir") then
+        return
+    end
+    if not checkIfBinExists("convert") then
+        return
+    end
+    if not checkIfBinExists("xdg-open") then
+        return
+    end
+    if not checkIfBinExists("xdg-user-dir") then
+        return
+    end
+
+    dt.print_error("Will try to export geoJSON file now")
+
+    local xportDirectory = dt.preferences.read("geoJSON_export","ExportDirectory","string")
+
+    -- Creates dir if not exsists
+    local imageFoldername = "files/"
+    local mkdirCommand = "mkdir -p "..exportDirectory.."/"..imageFoldername
+    coroutine.yield("RUN_COMMAND", mkdirCommand) 
+
+    -- Create the thumbnails
+    for _,image in pairs(image_table) do
```

Comment:
ususally _ is here to signal unused variable

if you actually use it, you should give it a proper name